### PR TITLE
NAS-127943 / 24.10 / Cloud backup API

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 6:00

File: src/middlewared/middlewared/plugins/cloud_backup/snapshot.py
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:70]: Have we tested what performance is like when dir contains 100K files? 1 million files? Does it need to be a job?

File: src/middlewared/middlewared/plugins/cloud_backup/restore.py
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cloud_backup/restore.py:17]: Are these local filesystem paths? Do we need to prevent people from doing silly things like overwriting `/etc`?

File: src/middlewared/middlewared/plugins/cloud_backup/snapshot.py
Comment (anodos325) [inline src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:30]: Have we tested under large snapshot counts to see whether this needs to be a job?

File: src/middlewared/middlewared/plugins/cloud_backup/snapshot.py
Comment (themylogin) [reply src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:70]: It's not a job in the original cloud sync plugin and it never caused any issues. People don't use this to list directories with 100k files (and we can't offer them to do so since no pagination is available).

File: src/middlewared/middlewared/plugins/cloud_backup/snapshot.py
Comment (themylogin) [reply src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:30]: 10000 snapshots take 30 seconds to list, I doubt any reasonable use will go above that.

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:30]: Are these 10k "restic" snapshots or zfs snapshots?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/cloud_backup/snapshot.py:30]: Restic snapshots, ZFS snapshots are not involved here

**Review by yocalebo — APPROVED** [review]
I'm good with it but wait for @anodos325 final approval before merge, please

**Pull request merged**
